### PR TITLE
Fix bug: Get mapped location of definition for findAllReferencesFull

### LINKED ISSUE #26175: [Project References] references-full returning definition in mapped .d.ts file
Setup: Function F declared in Project1, consumed in Project2.
Repro: Request `references-full` for the occurrence in Project 2.
Response (with unrelated properties pruned):
```json
{
  "seq": 0,
  "type": "response",
  "command": "references-full",
  "request_seq": 51,
  "success": true,
  "body": [
    {
      "definition": {
        "fileName": "C:/Users/amcasey/source/repos/NodeJsProjRef/NodeJsProjRef/bin/proj2.d.ts",
        "name": "function NS18.stuff(): void",
        "textSpan": {
          "start": 39,
          "length": 5
        }
      },
      "references": [
        {
          "fileName": "C:/Users/amcasey/source/repos/NodeJsProjRef/NodeJsProjRef/proj2/file2.ts",
          "textSpan": {
            "start": 45,
            "length": 5
          }
        }
      ]
    },
    {
      "definition": {
        "fileName": "C:/Users/amcasey/source/repos/NodeJsProjRef/NodeJsProjRef/proj1/file1.ts",
        "name": "function NS18.stuff(): void",
        "textSpan": {
          "start": 35,
          "length": 5
        }
      },
      "references": [
        {
          "fileName": "C:/Users/amcasey/source/repos/NodeJsProjRef/NodeJsProjRef/proj1/file1.ts",
          "textSpan": {
            "start": 35,
            "length": 5
          }
        }
      ]
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes #26175

### CODE REVIEW

**Comment by amcasey** [comment]
I think @uniqueiniquity was having some trouble in related scenarios.

**Comment by uniqueiniquity** [comment]
@amcasey I did, but I'm not sure whether I had my project configured correctly or not.

File: src/server/session.ts [hunk 1]
```diff
@@ -373,12 +380,17 @@ namespace ts.server {
...
+        getMappedLocation: (project: Project, location: sourcemaps.SourceMappableLocation) => sourcemaps.SourceMappableLocation | undefined,
+    ) => void;
+
     function combineProjectOutputWorker<TLocation extends sourcemaps.SourceMappableLocation | undefined>(
         projects: Projects,
         defaultProject: Project,
         initialLocation: TLocation,
         projectService: ProjectService,
-        cb: (where: ProjectAndLocation<TLocation>, getMappedLocation: (project: Project, location: sourcemaps.SourceMappableLocation) => boolean) => void,
+        cb: CombineProjectOutputCallback<TLocation>,
```
Comment (amcasey) [inline src/server/session.ts]: Personally, I find it confusing that some callers pass a callback with a different goal (tryAddToTodo) and no intention of returning.  Maybe the return type should also be a type parameter?

Comment (andy-ms) [reply src/server/session.ts]: You mean no intention of using the return value of `getMappedLocation`? Though every caller uses the return type of at least to test whether there was something added. It's true that they use it like a boolean but I don't think there's a type-safe way to have the return value be a type parameter that's either `boolean` or `sourcemaps.SourceMappableLocation | undefined`.

**Pull request merged**